contracts/libs/Ibc.sol

```diff
@@ -65,10 +65,16 @@ struct Height {
     uint64 revision_height;
 }
 
+enum AckStatus {
+    FAILURE,
+    SUCCESS,
+    SKIP
+}
+
 struct AckPacket {
     // success indicates the dApp-level logic. Even when a dApp fails to process a packet per its dApp logic, the
     // delivery of packet and ack packet are still considered successful.
-    bool success;
+    AckStatus status;
```

Comment:
I'm just now realizing that, because this `AckPacket` type is used in the acknowledgement proof, we can't change this existing type and need to add a new type with the enum. Otherwise we might have to make changes to peptide to support this new enum change. 

So I think we need to have two seperate `AckPackets` types now:
- an `AckPacket` with the enum status which the dapp returns in the onrecv callback
- the original `AckPacket` type with the bool success that is emitted in the dispatcher events (and in turn used in peptide proofs)